libs/cde-visualization/src/lib/cde-visualization/cde-visualization.component.ts

```diff
@@ -1,12 +1,182 @@
-import { ChangeDetectionStrategy, Component } from '@angular/core';
+import { ChangeDetectionStrategy, Component, computed, input } from '@angular/core';
 import { CommonModule } from '@angular/common';
+import { VisualizationHeaderComponent } from '../components/visualization-header/visualization-header.component';
+import { Metadata, MetadataComponent } from '../components/metadata/metadata.component';
+import { CellTypeOption, CellTypesComponent } from '../components/cell-types/cell-types.component';
+
+export interface Node {
+  x: number;
+  y: number;
+  z?: number;
+  cell_type: string;
+  cell_ontology_id?: string;
+}
+
+export interface ColorMapItem {
+  cell_type: string;
+  cell_color: string;
+}
+
+const DEFAULT_CELL_TYPE_ANCHOR = 'Endothelial';
+const DEFAULT_CELL_TYPE_COLOR = '#5D667F';
 
 @Component({
-  selector: 'cde-cde-visualization',
+  selector: 'cde-visualization',
   standalone: true,
-  imports: [CommonModule],
+  imports: [CommonModule, VisualizationHeaderComponent, MetadataComponent, CellTypesComponent],
   templateUrl: './cde-visualization.component.html',
   styleUrl: './cde-visualization.component.scss',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CdeVisualizationComponent {}
+export class CdeVisualizationComponent {
+  readonly nodes = input<Node[]>([
```

Comment:
We have to update this to work both with an array input or a file url input. It is probably easiest if we update this together over a zoom call.

Reply:
now works with both string and nodes[]